PowerEditor/src/MISC/Common/Common.cpp

```diff
@@ -911,3 +911,26 @@ bool matchInList(const TCHAR *fileName, const std::vector<generic_string> & patt
 	return false;
 }
 
+void createFullDirectory(const generic_string &dirName)
+{
+	TCHAR tmpName[MAX_PATH];
+
+	// Create a copy so we can modify it
+	_tcscpy_s(tmpName, MAX_PATH, dirName.c_str());
+
+	// Make sure the path ends in a backslash
+	if (dirName.back() != TEXT('\\'))
+		_tcscat_s(tmpName, MAX_PATH, TEXT("\\"));
+
+	// Create parent directories
+	for (auto p = _tcschr(tmpName, TEXT('\\')); p != NULL; p = _tcschr(p + 1, TEXT('\\')))
+	{
+		*p = 0;
+		// May or may not already exist
+		::CreateDirectory(tmpName, NULL);

```

Comment:
Instead calling CreateDirectory for each folder which already exist,
SHCreateDirectory can be used on fullpath. Whole logic of method createFullDirectory can be fallback option for SHCreateDirectory.

>MSDN remark for SHCreateDirectory -
This function creates a file system folder whose fully qualified path is given by pszPath. If one or more of the intermediate folders do not exist, it creates them.